### PR TITLE
Doctest info concepts

### LINKED ISSUE #24597: Follow up from ComponentInfo python exposure
Follow up on valid comments made against #24473 by @soininen see PR. 

Namely:

1. Python >=3 check
2. making examples testable via `testcode`, `testoutput` directives

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #24597

**Tester**
* Should mainly be possible via code review
* No harm in checking generated concept pages for style/consistency

### CODE REVIEW

**Comment by soininen** [comment]
@OwenArnold The concepts pages did not show up in the index page. I took the liberty of adding the missing `.. categories::` directives and fixing some style/consistency issues while at it. Hopefully this is OK for you.

**Comment by OwenArnold** [comment]
@soininen Thanks! Much appreciated!

File: Framework/PythonInterface/mantid/geometry/src/Exports/ComponentInfoPythonIterator.cpp [hunk 1]
```diff
@@ -19,10 +19,10 @@ void export_ComponentInfoPythonIterator() {
   // Export to Python
   class_<ComponentInfoPythonIterator>("ComponentInfoPythonIterator", no_init)
       .def("__iter__", objects::identity_function())
-#ifdef IS_PY3K
-      .def("__next__", &ComponentInfoPythonIterator::next)
-#else
+#ifdef IS_PY2K
```
Comment (soininen) [inline Framework/PythonInterface/mantid/geometry/src/Exports/ComponentInfoPythonIterator.cpp]: :+1:

**Review by soininen — APPROVED** [review]
This makes Mantid a bit more future-Python proof and the testable usage examples are a nice addition as well.

**Pull request merged**